tests/models/test_boundary_condition.py

```diff
@@ -102,3 +108,205 @@ def test_boundary_condition_mixin(t_end: int):
+        bounds = self.domain_boundary_sides(sd)
+        bc = pp.BoundaryCondition(
+            sd,
+            bounds.north + bounds.south + bounds.east + bounds.west,
+            "neu",
+        )
+        bc.is_neu[bounds.west] = False
+        bc.is_dir[bounds.west] = True
+        return bc
+
+    def bc_type_darcy_flux(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        return self.bc_type_scalar_dir_neu(sd=sd)
+
+    def bc_type_fourier_flux(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        return self.bc_type_scalar_dir_neu(sd=sd)
+
+
+class MassBalanceNeu(
+    SquareDomainOrthogonalFractures,
+    BCValues,
+    BCDirNeu,
+    pp.models.fluid_mass_balance.SinglePhaseFlow,
+): ...
+
+
+class MassBalanceRob(BCRobDir, MassBalanceNeu): ...
+
+
+class MassAndEnergyBalanceNeu(
+    SquareDomainOrthogonalFractures,
+    BCValues,
+    BCDirNeu,
+    pp.models.mass_and_energy_balance.MassAndEnergyBalance,
+): ...
+
+
+class MassAndEnergyBalanceRob(BCRobDir, MassAndEnergyBalanceNeu): ...
+
+
+class MomentumBalanceNeu(
+    SquareDomainOrthogonalFractures,
+    BCValues,
+    BCDirNeu,
+    pp.models.momentum_balance.MomentumBalance,
+): ...
+
+
+class MomentumBalanceRob(BCRobDir, MomentumBalanceNeu): ...
+
+
+@pytest.fixture()
+def run_models():
+    params = {
+        "times_to_export": [],
+        "fracture_indices": [],
+        "meshing_arguments": {"cell_size": 0.5},
+    }
+    models = {}
+
+    def run_model(balance_class):
+        instance = balance_class(params)
+        pp.run_time_dependent_model(instance, params)
+        sd = instance.mdg.subdomains(dim=2)[0]
+
+        if isinstance(instance, (MassBalanceRob, MassBalanceNeu)):
+            pressure = instance.pressure([sd]).value(instance.equation_system)
+            return {"pressure": pressure}
+        elif isinstance(instance, (MomentumBalanceRob, MomentumBalanceNeu)):
+            displacement = instance.displacement([sd]).value(instance.equation_system)
+            return {"displacement": displacement}
+        elif isinstance(instance, (MassAndEnergyBalanceRob, MassAndEnergyBalanceNeu)):
+            temperature = instance.temperature([sd]).value(instance.equation_system)
+            return {"temperature": temperature}
+
+    models["mass_balance"] = {
+        "rob": run_model(MassBalanceRob),
+        "neu": run_model(MassBalanceNeu),
+    }
+
+    models["momentum_balance"] = {
+        "rob": run_model(MomentumBalanceRob),
+        "neu": run_model(MomentumBalanceNeu),
+    }
+
+    models["mass_and_energy_balance"] = {
+        "rob": run_model(MassAndEnergyBalanceRob),
+        "neu": run_model(MassAndEnergyBalanceNeu),
+    }
+    return models
+
+
+@pytest.mark.parametrize(
+    "balance_type", ["mass_balance", "momentum_balance", "mass_and_energy_balance"]
+)
+def test_robin_limit_case(run_models, balance_type):
+    """Test that Robin conditions are equivalent to Neumann with Robin weight = 0.
```

Comment:
Nice documentation. Thanks!

Reply:
In case I am not mistaken, you are testing one limit only, alpha -> 0 (Neuman). However, I believe that it would be relevant to include a test for alpha -> inf (Dirichlet), numerically large values {1e+12, 1e+14, 1e+16}.

Reply:
Makes sense to include both. I am not sure if I will be able to finish it before I go off on holidays, but I have it on my to do list.